prov/mrail/src/mrail_cq.c

```diff
@@ -383,8 +383,12 @@ void mrail_poll_cq(struct util_cq *cq)
 			tx_buf = comp.op_context;
 
 			if (tx_buf->flags & FI_COMPLETION) {
-				ret = ofi_cq_write(cq, comp.op_context,
-						   comp.flags, 0, NULL, 0, 0);
+				tx_buf->flags = ((tx_buf->flags &
+						  (FI_TAGGED | FI_MSG)) |
+						 FI_SEND);
+				ret = ofi_cq_write(cq, tx_buf->context,
+						   tx_buf->flags,
+						   0, NULL, 0, 0);
```

Comment:
You can pass (tx_buf->flags & (FI_TAGGED | FI_MSG)) | FI_SEND directly into ofi_cq_write() and avoid the extra memory write.